### DISCUSSION TITLE
How to 'ensureuser' in batch?

### DISCUSSION THREAD

**Discussion by kongmengfei** [pnp/pnpcore#899] [Q&A]
I am using pnpcore sdk. i have a string Array with 10+ user UPN in my C# project. now i want to resolve these users. i find below method:

[EnsureUserBatchAsync](https://pnp.github.io/pnpcore/api/PnP.Core.Model.SharePoint.IWeb.html#collapsible-PnP_Core_Model_SharePoint_IWeb_EnsureUserBatch_System_String_)

![image](https://user-images.githubusercontent.com/57654821/176152703-dcc0f7eb-7d61-46b3-b075-157b80214388.png)

i am trying to ensure these users in batch, try to use below code but still need help

![image](https://user-images.githubusercontent.com/57654821/176153024-1a7dbd38-3f4c-48c3-a634-93f3ca54bdf4.png)

do i need to call [ExecuteAsync](https://pnp.github.io/pnpcore/api/PnP.Core.Services.PnPContext.html#collapsible-PnP_Core_Services_PnPContext_ExecuteAsync_System_Boolean_) ?

How to receive the result? and how to handle the exception if there is no such user?

help me please 😭

**Comment by jansenbe** [comment]
Yes, you need to call `ExecuteAsync`. See https://pnp.github.io/pnpcore/using-the-sdk/basics-batching.html for details on batching and how to handle the exceptions
